apps/site/src/lib/components/Navbar/Navbar.tsx

```diff
@@ -0,0 +1,87 @@
+"use client";
+
+import * as NavMenu from "@radix-ui/react-navigation-menu";
+
+import styles from "./Navbar.module.scss";
+import React from "react";
+import { useState, useEffect } from "react";
+
+import NavLinkItem from "./NavbarHelpers";
+import Button from "@/lib/components/Button/Button";
+import HackLogo from "@/lib/components/HackLogo/HackLogo";
+
+import hamburger from "@/assets/icons/navigation-icon.svg";
+import Image from "next/image";
+
+function Navbar() {
+	const [listShown, setListShown] = useState(false);
+	const [transitionApplied, setTransitionApplied] = useState(false);
+	const [hasScrolled, setHasScrolled] = useState(false);
+
+	useEffect(() => {
+		const mobileNavInitStateHandler = () => {
+			setListShown(false);
+			setTransitionApplied(false);
+		};
+
+		// hides list on breakpoint
+		window
+			.matchMedia("(min-width: 768px)")
+			.addEventListener("change", mobileNavInitStateHandler);
```

Comment:
Nitpick [repeating]: For [`useEffect`](https://react.dev/reference/react/useEffect), you'll want to return a function that removes the event listener when this component is unmounted.

Reply:
Suggestion: As Taesung noted, it might be easier to use some media queries and `display: hidden/block` instead of the event listener.